### PR TITLE
Fix graph creation to avoid using migration tags

### PR DESCRIPTION
**Pull request by jesica-fera**
Also, recreate the graph every time so the XML references are correct

### CODE REVIEW

File: mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/applicationgraph/ApplicationGraphCreator.java [hunk 1]
```diff
@@ -103,17 +99,21 @@ private MessageProcessor convertToComponent(Element xmlElement, Flow parentFlow,
       } else {
         report.report("nocompatibility.dynamicflowref", xmlElement, xmlElement);
       }
-    } else if (isOperation(xmlElement)) {
-      return new MessageOperation(xmlElement, parentFlow);
+    } else if (isPropertySource(xmlElement, parentFlow)) {
+      if (isFirstElement(xmlElement, parentFlow)) {
+        return new MessageSource(xmlElement, parentFlow);
```
Comment (svacas) [inline mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/applicationgraph/ApplicationGraphCreator.java:104]: Being the first element in the flow does not necessary mean that it is a message source.
A flow without source could have an `http:request` as the first element for instance.

File: mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/nocompatibility/TranslateInboundReferencesStep.java [hunk 2]
```diff
@@ -65,36 +65,39 @@ public String getDescription() {
...
-        while (!elementsToUpdate.isEmpty()) {
-          Element element = elementsToUpdate.poll();
-          // update references in attributes
-          updatePropertyReferencesInAttributes(element, propertiesEmitter.get().getType(), report);
-          Optional<CDATA> cdataContent = getCDATAContent(element.getContent());
-          if (element.getChildren().isEmpty() && cdataContent.isPresent()) {
-            updatePropertyReferencesInCDATAContent(element, cdataContent.get(), propertiesEmitter.get().getType(), report);
-          } else {
-            element.getChildren().forEach(e -> elementsToUpdate.offer(e));
+  public void execute(Element unused, MigrationReport report) throws RuntimeException {
```
Comment (svacas) [inline mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/nocompatibility/TranslateInboundReferencesStep.java:68]: Not required for this task, but this step should be removed and the translation logic be part of the expression migrator

Comment (jesica-fera) [reply mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/nocompatibility/TranslateInboundReferencesStep.java:68]: there is a task in the sprint already to do this, I just didn't want to tackle everything in the same PR

File: mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/applicationgraph/ApplicationGraphCreator.java (hunk 1 above)
Comment (jesica-fera) [reply mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/applicationgraph/ApplicationGraphCreator.java]: I just removed the 2 classes (MessageSource and MessageOperation) and created a single implementation

**Pull request merged**